torch_xla/csrc/tensor.h

```diff
@@ -270,12 +242,15 @@ class XLATensor : public torch::lazy::LazyTensor {
   torch::lazy::Value GetIrValueForTensor(
       const at::Tensor& tensor, const torch::lazy::BackendDevice& device) const;
 
-  static int64_t GetNextTensorId();
-
   static bool UseEagerDebugMode();
 
   bool ShouldSyncIrNode();
 
+  // We store two shared_ptr of Data in a XLATensor.
+  // One in the LazyTensor class as the LazyTensor::Data type
+  // for base class method to access. One here as the derived
+  // XLATensor::Data type such that it's easier to access XLA
+  // extra fields.
```

Comment:
shouldn't `XLATensor::Data` inherits `torch::lazy::Data` and we only need to access `XLATensor::Data`?

Reply:
Base class has methods like GetUniqueId() which will need to access data_, and therefore we need to set the data_ field there. However, we also need to provide a data() accessor that returns a const reference to the data_ field but with XLATensor::Data type. That's why we also need to store another pointer within XLATensor.

Reply:
I guess after we remove the view, `logical_element_type` is the only thing unique in our `Data`?

Reply:
OK so, it seems like we only create one `Data` object which is a `XLATensor::Data` but we have two pointers of different type point to this object. Is this a fair understanding?

Reply:
That's correct!